### PR TITLE
fix reshape in Mnist Datasets

### PR DESCRIPTION
**Pull request by kojix2**

### CODE REVIEW

**Comment by kojix2** [comment]
I changed pull request.

File: lib/chainer/datasets/mnist.rb [hunk 1]
```diff
@@ -15,9 +15,9 @@ def self.get_mnist(withlabel: true, ndim: 1, scale: 1.0, dtype: Numo::DFloat, la
       def self.preprocess_mnist(raw, withlabel, ndim, scale, image_dtype, label_dtype)
         images = raw[:x]
         if ndim == 2
-          images = images.reshape(-1, 28, 28)
+          images = images.reshape(nil, 28, 28)
         elsif ndim == 3
-          images = images.reshape(-1, 1, 28, 28)
+          images = images.reshape(nil, 1, 28, 28)
```
Comment (hatappi) [inline lib/chainer/datasets/mnist.rb]: I'd like to unify either `nil` or `true`.
Currently `true` is used.
Is there a better reason for `nil`?

https://github.com/red-data-tools/red-chainer/blob/master/lib/chainer/functions/loss/softmax_cross_entropy.rb#L45

**Comment by hatappi** [comment]
@kojix2 
Thank you p-r!!
I comment this p-r.
Please check comment 🙏

File: lib/chainer/datasets/mnist.rb (hunk 1 above)
Comment (kojix2) [reply lib/chainer/datasets/mnist.rb]: I'm sorry. I misunderstood that "nil" is used instead of "True" in this project.
I think that "true" is good.

**Comment by hatappi** [comment]
@kojix2 
Thank you for fixing!!

**Pull request merged**